From: Eric Dumazet <eric.dumazet@gmail.com>
Date: Thu, 13 Oct 2011 08:03:36 +0000
Subject: e1000e: fix skb truesize underestimation
Patch-mainline: v3.2-rc1
Git-commit: 98a045d7e4a59db0865a59eea2140fe36bc7c220
References: bnc#795174 FATE#313648

e1000e allocates a page per skb fragment. We must account
PAGE_SIZE increments on skb->truesize, not the actual frag length.

Signed-off-by: Eric Dumazet <eric.dumazet@gmail.com>
CC: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/ethernet/intel/e1000e/netdev.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)
--- a/drivers/net/ethernet/intel/e1000e/netdev.c
+++ b/drivers/net/ethernet/intel/e1000e/netdev.c
@@ -1316,7 +1316,7 @@ static bool e1000_clean_rx_irq_ps(struct
 			ps_page->page = NULL;
 			skb->len += length;
 			skb->data_len += length;
-			skb->truesize += length;
+			skb->truesize += PAGE_SIZE;
 		}
 
 		/* strip the ethernet crc, problem is we're using pages now so
@@ -1376,7 +1376,7 @@ static void e1000_consume_page(struct e1
 	bi->page = NULL;
 	skb->len += length;
 	skb->data_len += length;
-	skb->truesize += length;
+	skb->truesize += PAGE_SIZE;
 }
 
 /**
